server/src/main/java/io/spine/examples/kanban/server/board/BoardAggregate.java

```diff
@@ -176,8 +173,134 @@ private void apply(ColumnPlaced e) {
+     */
+    private boolean isTotalActual(ColumnPosition p) {
+        return p.getOfTotal() == state().getColumnCount();
+    }
+
+    /**
+     * Checks whether the provided column is actually placed at the index from
+     * the provided position.
+     *
+     * @return {@code true} if the column is placed at the index from the provided position.
+     */
+    private boolean isIndexActual(ColumnId c, ColumnPosition p) {
+        return p.zeroBasedIndex() == state().getColumnList().indexOf(c);
+    }
+
+    /**
+     * Shifts columns to fill the void left by the moving column.
+     *
+     * <p> The shift direction is based on the movement direction. If the column is
+     * moving right, then the columns on the way are shifted left and vice versa.
+     */
+    private ImmutableList<ColumnMovedOnBoard> shiftColumns(MoveColumn c) {
+        if (isColumnMovingRight(c.getFrom(), c.getTo())) {
+            ColumnPosition rightToFrom = rightTo(c.getFrom());
+            return shiftColumnsLeft(rightToFrom, c.getTo());
+        } else {
+            ColumnPosition leftToFrom = leftTo(c.getFrom());
+            return shiftColumnsRight(leftToFrom, c.getTo());
+        }
+    }
+
+    private static boolean isColumnMovingRight(ColumnPosition from, ColumnPosition to) {
+        return from.getIndex() < to.getIndex();
+    }
+
+    /**
+     * Returns the position right to the provided one.
+     *
+     * <p> The total number of columns remains unchanged.
+     */
+    private static ColumnPosition rightTo(ColumnPosition p) {
+        return ColumnPositions.of(p.getIndex() + 1, p.getOfTotal());
+    }
+
+    /**
+     * Returns the position left to the provided one.
+     *
+     * <p> The total number of columns remains unchanged.
+     */
+    private static ColumnPosition leftTo(ColumnPosition p) {
+        return ColumnPositions.of(p.getIndex() - 1, p.getOfTotal());
+    }
+
+    private ImmutableList<ColumnMovedOnBoard> shiftColumnsLeft(ColumnPosition from, ColumnPosition to) {
+        ImmutableList.Builder<ColumnMovedOnBoard> movedColumns = new ImmutableList.Builder<>();
+
+        ColumnPosition current = from;
+        while (!current.equals(to)) {
+            ColumnPosition newPosition = leftTo(current);
+            movedColumns.add(moveColumn(current, newPosition));
+            current = rightTo(current);
+        }
+
+        return movedColumns.build();
+    }
+
+    private ImmutableList<ColumnMovedOnBoard> shiftColumnsRight(ColumnPosition from, ColumnPosition to) {
+        ImmutableList.Builder<ColumnMovedOnBoard> movedColumns = new ImmutableList.Builder<>();
+
+        ColumnPosition current = from;
+        while (!current.equals(to)) {
+            ColumnPosition newPosition = rightTo(current);
+            movedColumns.add(moveColumn(current, newPosition));
+            current = leftTo(current);
+        }
+
+        return movedColumns.build();
```

Comment:
How about extracting the logic of column movement (and all the checks) into some nested type? 

The idea is to use a so-called method object when the enclosing type (the type of this Aggregate, in this case) gets overwhelmed by lots of methods, devoted to some operation.

When applied, such an approach improves the readability and documentability of both the enclosing type, and the type of "method object".